### PR TITLE
Do clippy improvements

### PR DESCRIPTION
**Pull request by tcharding**
Clippy has been updated and new warnings are being triggered in our codebase. This PR does all warnings using nightly since they all looked like reasonable things to fix.

Needed for CI to pass in other open PRs.

### CODE REVIEW

File: bitcoin/src/bip152.rs [hunk 1]
```diff
@@ -284,7 +284,7 @@ impl Decodable for BlockTransactionsRequest {
                 // Since the number of indices ultimately represent transactions,
                 // we can limit the number of indices to the maximum number of
                 // transactions that would be allowed in a vector.
-                let byte_size = (nb_indexes as usize)
+                let byte_size = (nb_indexes)
```
Comment (Kixunil) [inline bitcoin/src/bip152.rs:287]: Hmm, it didn't complain about useless parentheses?

**Review by Kixunil — APPROVED** [review]
ACK 1050fe9cae88643ebe18436bc99a02f2c1c9961a

Nice cleanups!

**Review by sanket1729 — APPROVED** [review]
ACK 1050fe9cae88643ebe18436bc99a02f2c1c9961a.

File: bitcoin/src/bip152.rs (hunk 1 above)
Comment (sanket1729) [reply bitcoin/src/bip152.rs:287]: We should remove this later. Merging this to get the CI rolling

Comment (tcharding) [reply bitcoin/src/bip152.rs:287]: bah, I must be blind. Will fix, thanks.

**Pull request merged**